pkg/sip/inbound.go

```diff
@@ -72,37 +83,93 @@ func (s *Server) getInvite(from string) *inProgressInvite {
 }
 
 func (s *Server) handleInviteAuth(log logger.Logger, req *sip.Request, tx sip.ServerTransaction, from, username, password string) (ok bool) {
+	log = log.WithValues(
+		"from", from,
+		"username", username,
+		"passwordHash", hashPassword(password),
+		"hasPassword", password != "",
+		"method", req.Method.String(),
+		"uri", req.Recipient.String(),
+	)
+
+	log.Infow("Starting SIP invite authentication")
+
 	if username == "" || password == "" {
+		log.Infow("Skipping authentication - no credentials provided")
 		return true
 	}
+
 	if s.conf.HideInboundPort {
 		// We will send password request anyway, so might as well signal that the progress is made.
+		log.Debugw("Sending processing response due to HideInboundPort config")
 		_ = tx.Respond(sip.NewResponseFromRequest(req, 100, "Processing", nil))
 	}
 
 	inviteState := s.getInvite(from)
+	log = log.WithValues("inviteStateFrom", from)
 
 	h := req.GetHeader("Proxy-Authorization")
 	if h == nil {
-		log.Infow("Requesting inbound auth")
+		log.Infow("Requesting inbound auth - no Proxy-Authorization header found")
 		inviteState.challenge = digest.Challenge{
 			Realm:     UserAgent,
 			Nonce:     fmt.Sprintf("%d", time.Now().UnixMicro()),
 			Algorithm: "MD5",
 		}
 
+		log.Debugw("Created digest challenge",
+			"realm", inviteState.challenge.Realm,
+			"nonce", inviteState.challenge.Nonce,
+			"algorithm", inviteState.challenge.Algorithm,
+		)
+
 		res := sip.NewResponseFromRequest(req, 407, "Unauthorized", nil)
 		res.AppendHeader(sip.NewHeader("Proxy-Authenticate", inviteState.challenge.String()))
 		_ = tx.Respond(res)
+		log.Infow("Sent 407 Unauthorized response with Proxy-Authenticate header")
 		return false
 	}
 
+	log.Debugw("Found Proxy-Authorization header, parsing credentials")
 	cred, err := digest.ParseCredentials(h.Value())
 	if err != nil {
+		log.Errorw("Failed to parse Proxy-Authorization credentials", err,
+			"headerValue", h.Value(),
+		)
+		_ = tx.Respond(sip.NewResponseFromRequest(req, 401, "Bad credentials", nil))
+		return false
+	}
+
+	log.Debugw("Parsed credentials successfully",
+		"credUsername", cred.Username,
+		"credURI", cred.URI,
+		"credRealm", cred.Realm,
```

Comment:
You could also just dump the whole struct as `cred`. But I haven't looked if there are sensitive fields there or not.